Source/hydro/Castro_mol.cpp

```diff
@@ -6,10 +6,260 @@
 #include "Radiation.H"
 #endif
 
+#ifdef GRAVITY
+#include "Gravity.H"
+#endif
+
 #include <ppm.H>
 
 using namespace amrex;
 
+void
+Castro::mol_plm_reconstruct(const Box& bx,
+                            const int idir,
+                            Array4<Real const> const q_arr,
+                            Array4<Real const> const flatn_arr,
+                            Array4<Real> const dq,
+                            Array4<Real> const qm,
+                            Array4<Real> const qp) {
+
+  const auto dx = geom.CellSizeArray();
+
+#ifdef GRAVITY
+  Real lconst_grav =  gravity->get_const_grav();
+#else
+  Real lconst_grav = 0.0_rt;
+#endif
+
+  AMREX_PARALLEL_FOR_4D(bx, NQ, i, j, k, n,
+  {
+
+   // piecewise linear slopes
+   uslope(bx, idir,
+          q_arr, n,
+          flatn_arr, dq);
+
+
+   // this is a loop over zones.  For each slope in the zone, fill the
+   // two adjacent edge states (e.g., the right state at i-1/2 and the
+   // left state at i+1/2
+
+   if (idir == 0) {
+
+     if (plm_well_balanced == 1 && n == QPRES && idir == AMREX_SPACEDIM-1) {
```

Comment:
For CUDA we'll need local copies of `plm_well_balanced`.

Reply:
yes